### PR TITLE
C++/C#: Move the Windows autobuilder into a subfolder in `ql/csharp`.

### PR DESCRIPTION
**Pull request by criemen**
This is a necessary preparation for moving the C# dependency management to `paket`, which in turn is a necessary preparation for moving the C# build to bazel.

As we discovered in https://github.com/github/codeql/pull/16376, `paket` tries to restore all projects recursively from the root folder. If we support building C# code under both `ql/csharp` and `ql/cpp`, we need to have a single lockfile under `ql`, as both codebases share the same set of dependencies (and utilities from `ql/csharp/extractor`).
Then, `paket` will also try to restore things that look like "C# projects" in other languages' folders, which is not what we want.
Therefore, we address this by moving all C# code into a common root directory, `ql/csharp`.

This needs an internal PR to adjust the buildsystem to look for the autobuilder in the new location.

### CODE REVIEW

**Review by tamasvajk — DISMISSED** [review]
LGTM

**Comment by criemen** [comment]
As discussed in the other PR, waiting for explicit approval from @github/codeql-c-extractor , who are currently at an offsite.

File: CODEOWNERS
Comment (jketema) [inline CODEOWNERS:2]:
```suggestion
```
as the folder will cease to exist.

**Review by jketema — COMMENTED** [review]
Change-wise this LGTM, just one little comment.

Before approving this I'd like to understand two things related to releases:
* Will the C++ Windows autobuilder still be included?
* Will the C++ Windows autobuilder still be included in such a way that it is correctly picked up when attempting autobuild of a C/C++ project on Windows?

File: CODEOWNERS
Comment (criemen) [reply CODEOWNERS:2]: Well in theory the README file stays there, but you're right, it doesn't require a separate CODEOWNERS entry. I'll delete the line.

**Comment by criemen** [comment]
Re releases: The internal PR changes the path where we're looking for the autobuilder, so that it's still correctly included in the released bundle. I verified that manually with a bundle build. The invocation of the autobuilder doesn't change at all by the moving of the directories.

**Review by jketema — APPROVED** [review]

**Pull request merged**
